WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -89,10 +90,16 @@ private void handleIntent(Intent intent) {
             actionBar.setDisplayHomeAsUpEnabled(true);
         }
 
+        PostModel targetPost = null;
+        int targetPostId = intent.getIntExtra(EXTRA_TARGET_POST_ID, 0);
+        if (targetPostId > 0) {
+            targetPost = mPostStore.getPostByLocalPostId(intent.getIntExtra(EXTRA_TARGET_POST_ID, 0));
+        }
+
         mPostList = (PostsListFragment) getFragmentManager().findFragmentByTag(PostsListFragment.TAG);
-        if (mPostList == null || siteHasChanged || pageHasChanged) {
+        if (mPostList == null || siteHasChanged || pageHasChanged || targetPost != null) {
             PostsListFragment oldFragment = mPostList;
-            mPostList = PostsListFragment.newInstance(mSite, mIsPage);
+            mPostList = PostsListFragment.newInstance(mSite, mIsPage, targetPost);
```

Comment:
`PostsListFragment.newInstance()`  only uses `targetPost` to obtain it's `id` and set a parcelable argument. Later on that `id` will be properly used to retrieve the actual `PostModel` from the db when it is actually needed.
Also, the only reason why `PostStore` is imported and injected here is to be able to get the PostModel.

Why don't we just remove the use of `mPostStore` in PostsListActivity altogether, and pass the ID here as well?

This would also be in line with the decision taken on trying to avoid this kind of `PostModel` passing around because of potential OOM problems, like in this PR here https://github.com/wordpress-mobile/WordPress-Android/pull/5856